tools/pepquery/pepquery.xml

```diff
@@ -143,39 +155,63 @@
                 </when>
                 -->
             </conditional>
-            <param name="db_file" type="data" format="fasta" label="Protein Reference Database File" argument="-db" help="an input sequence that matches a reference will be ignored." />
-            <param name="spectrum_file" type="data" format="mgf" label="Spectrum File" argument="-ms" help="Spectrum file used for identification, mgf format" />
-            <param name="indexType" type="select" label="Report Spectrum Scan as" argument="-indexType" help="" >
+            <param name="db_file" argument="-db" type="data" format="fasta" label="Protein Reference Database File" help="an input sequence that matches a reference will be ignored." />
+            <param name="spectrum_file" argument="-ms" type="data" format="mgf" label="Spectrum File" help="Spectrum file used for identification, mgf format" />
+            <param name="indexType" argument="-indexType" type="select" label="Report Spectrum Scan as" help="" >
                 <option value="1" selected="true">index (1-based) in MGF</option>
                 <option value="2">spectrum title in MGF</option>
             </param>
+            <conditional name="tags">
+                <param name="tagType" type="select" label="Tag File">
+                    <option value="None">None</option>
+                    <option value="PepQuery">PepQuery</option>
+                    <option value="pFind">pFind</option>
+                </param>
+                <when value="None"/>
+                <when value="PepQuery">
+                    <param name="tag_file" argument="-tag" type="data" format="txt" label="PepPuery Tag File"/>
+                    <param name="tagIndexType" argument="-tagIndexType" type="select" label="Tag File Index Type" help="pFind uses the spectrum title" >
+                        <option value="1">index (1-based) in MGF</option>
+                        <option value="2" selected="true">spectrum title in MGF</option>
+                    </param>
+                </when>
+                <when value="pFind">
```

Comment:
What is this pFind?

Reply:
@bgruening   http://pepquery.org/data/PepQuery_for_immunopeptidomics_data.pdf 
Open-pFind can be downloaded from http://pfind.ict.ac.cn/software/pFind3/. It can only be run on Windows system.  
pFind generates peptide tags that can be used to greatly reduce the search space for pepquery when searching without an enzyme cleavage.   Someone suggested that MsFragger might also help in that regard, but I don't have the knowledge to evaluate that proposition.